File: frontend/src/types/UserQuery.ts
export type UserQuery = {
    id : number;
    userName : string;
    email : string;
    username : string;
    password : string;
    login : string;
    languages : string;
    role : string;
    dateInfo : string;
};

/*
"fullName": "string",
  "email": "string",
  "login": "string",
  "languages": [
    {
      "language": "string",
      "level": "string"
    }
  ],
  "role": "string",
  "createdAt": "2024-08-01T14:34:09.843Z",
  "deletedAt": "2024-08-01T14:34:09.843Z"
*/

export type UserCreateQuery = {
    fullName? : string;
    email : string;
    login? : string;
    languages? : string;
    role? : string;
    createdAt? : string;
    deletedAt? : string;
}

File: backend/project-service/src/main/java/ru/itpark/projectservice/infrastructure/authservice/dto/query/UserQuery.java
package ru.itpark.projectservice.infrastructure.authservice.dto.query;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.itpark.projectservice.infrastructure.authservice.valueobjects.DateInfo;
import ru.itpark.projectservice.infrastructure.authservice.valueobjects.Language;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserQuery {
    private Long id;
    private String fullName;
    private String email;
    private String username;
    private String password;
    private String login;
    private List<Language> languages;
    private String role;
    private DateInfo dateInfo;
}
